### ISSUE TITLE
Bug: TDEngine Version 3.3.4.8 - range fill(prev) Yields Output with Invalid Characters

### ISSUE THREAD

**Issue by mesquita** [taosdata/TDengine#29353]
**Bug Description**

We are having an issue when querying the database using range combined with fill(prev) where the end range timestamp extends beyond the last occurrence of a tag sample.

When trying to make a query like this:
```
select _irowts as `time`,
    `tag`,
    interp(`value`) as `value`
from sensors
where `tag` in ('ABC.PV')
partition by tbname
range('2024-12-01 00:00:00', '2025-01-01 00:00:00') every(600s) fill(prev);
```

the current output in this version has for the output `tag` the following value:
```
ABC.��
```

Instead of returning the expected value ABC.PV, the tag field contains invalid characters such as \x00\x00 when the query includes a “future” value (i.e., a timestamp after the last sample for the tag).

**To Reproduce**
1.	Run a query similar to the example above.
2.	Ensure the range includes an end timestamp beyond the last known sample for the tag.

**Expected Behavior**
The output should correctly display ABC.PV for the tag field if the interpolation value (interp) is correctly computed.

**Screenshots**
![image](https://github.com/user-attachments/assets/e5d9c015-bf4e-4985-8ec6-9b1acba09de8)

**Environment (please complete the following information):**
TDEngine version 3.3.4.8

Thank you very much!

**Comment by yu285** [comment]
let me check on this

**Comment by itepifanio** [comment]
Hi @yu285, thank you for taking the time to address this! It seems that the issue hasn't been resolved in the latest release. Could you share any updates or progress on this?

**Comment by yu285** [comment]
taosdump -o `path`  `databasename` sensors

**Comment by mesquita** [comment]
Hi @yu285, sure, here's a representative dump of the database: [Google Drive Link](https://drive.google.com/file/d/1BAhWTY8ML4tS-k6TBg1hZgr5kTYgGAsn/view?usp=sharing) or [WeTranfer Link](https://we.tl/t-fsB8nS8Usy).

You can reproduce the bug by running:

```
select _irowts as `time`, `tag`, interp(`value`) as `value` 
from sensors where `tag` in ('TAG-1234-01.PV') 
partition by tbname 
range('2024-12-01 00:00:00', '2025-01-01 00:00:00') 
every(600s) fill(prev);
```

Thank you sir!

**Comment by mesquita** [comment]
Hi @yu285, do you have any news on this one? Thanks!!

**Comment by yu285** [comment]
> Hi [@yu285](https://github.com/yu285), do you have any news on this one? Thanks!!

checking

**Comment by yu285** [comment]
we are fixing this ,will fix it in the next version  in the end of the Feb.

or you can compile it by yourself once the PR is merged 

we will inform you here .

**Comment by mesquita** [comment]
@yu285 thank you sir!

**Comment by yu285** [comment]
@mesquita  please use 3.3.5.2 to verify this issue

**Comment by mesquita** [comment]
@yu285, no, version 3.3.5.2 didn't fix the problem. I'm still having the same issue:

![Image](https://github.com/user-attachments/assets/eda4862f-5e8d-4418-8c5c-c715cf68bdb3)

**Comment by yu285** [comment]
let me check again

**Comment by yu285** [comment]
![Image](https://github.com/user-attachments/assets/e7be3b99-56c2-48d7-9e1e-ee1d650edb4e)

sorry it should be 3.3.5.4 ,my fault .

you can wait 3.3.5.4 released or compile newest main branch

**Comment by mesquita** [comment]
Thank you. I'll wait for the release of this version.

**Comment by mesquita** [comment]
Hello @yu285, we tested version 3.3.5.4 building it locally and the bug if fixed! Thank you very much sir! We're just waiting for the new version to be released to start using it.

**Comment by yu285** [comment]
> Hello [@yu285](https://github.com/yu285), we tested version 3.3.5.4 building it locally and the bug if fixed! Thank you very much sir! We're just waiting for the new version to be released to start using it.

that sounds good ~

**Comment by mesquita** [comment]
hi @yu285, do you have any expectations for the next release date?
